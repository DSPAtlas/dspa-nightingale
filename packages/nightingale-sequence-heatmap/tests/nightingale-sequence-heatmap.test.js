import "../dist/index";
import { xDomain, yDomain, data } from "./mockData/mockdata";

let rendered;

describe("nightingale-sequence-heatmap tests", () => {
  beforeEach(async () => {
    document.documentElement.innerHTML = `<nightingale-sequence-heatmap  heatmap-id="testId" width="500" height="500"></nightingale-sequence-heatmap>`;
    rendered = document.querySelector("nightingale-sequence-heatmap");
    await new Promise((resolve) => requestAnimationFrame(() => resolve()));
    await rendered.updateComplete;
    rendered.setHeatmapData(xDomain, yDomain, data);
  });

  afterEach(() => {
    document.querySelector("nightingale-sequence-heatmap").remove();
  });

  test("it should display the sequence heatmap correctly", async () => {
    expect(rendered).toMatchSnapshot();
  });

  test("it should zoom in", (done) => {
    rendered.setAttribute("display-start", "2");
    rendered.setAttribute("display-end", "4");
    window.requestAnimationFrame(() => {
      expect(rendered).toMatchSnapshot();
      done();
    });
  });

  test("it should display the sequence correctly after highlight", (done) => {
    rendered.setAttribute("highlight", "1:3");
    window.requestAnimationFrame(() => {
      expect(rendered).toMatchSnapshot();
      done();
    });
  });
});
