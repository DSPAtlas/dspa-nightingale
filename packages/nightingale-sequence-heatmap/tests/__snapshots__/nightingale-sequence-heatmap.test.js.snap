// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-sequence-heatmap tests it should display the sequence correctly after highlight 1`] = `
<nightingale-sequence-heatmap
  display-end="-1"
  display-start="1"
  heatmapid="testId"
  height="500"
  highlight="1:3"
  highlight-color="#FFEB3B66"
  min-height="10"
  min-width="10"
  style="display: inline-block; line-height: 0;"
  width="500"
>
  <!---->
  
      
  <style>
    
        #
    <!---->
    testId
     {
          /** Position of bottom-left corner of tooltip box relative to the mouse position */
          --tooltip-offset-x: 5px;
          /** Position of bottom-left corner of tooltip box relative to the mouse position */
          --tooltip-offset-y: 8px;
        }
        .heatmap-marker-x {
          /** background: 
    <!---->
    #FFEB3B66
     !important; */
          fill: 
    <!---->
    rgb(255, 235, 59)
     !important;
          fill-opacity: 
    <!---->
    0.359
     !important;
          stroke-width: 0 !important;
        }
        .heatmap-marker-y {
          /** background: 
    <!---->
    #FFEB3B66
     !important; */
          fill: 
    <!---->
    rgb(255, 235, 59)
     !important;
          fill-opacity: 
    <!---->
    0.359
     !important;
          stroke-width: 0 !important;
        }
        
    <!---->
    
.heatmap-main-div {
    font-family: sans-serif;
}

.heatmap-canvas-div {
    /* Set background-color here to change the background of the heatmap */
    background-color: none;
}

.heatmap-tooltip-box,
.heatmap-pinned-tooltip-box {
    z-index: 0;
    /* Avoid tooltip flickering */
    pointer-events: none;
}

.heatmap-tooltip-content,
.heatmap-pinned-tooltip-content {
    margin-left: var(--tooltip-offset-x);
    margin-bottom: var(--tooltip-offset-y);
    border: solid black 1px;
    padding-block: 0.35em;
    padding-inline: 0.7em;
    background-color: white;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.75);
    pointer-events: initial;
    width: max-content;
    line-height: 1;
}

.heatmap-pinned-tooltip-close {
    position: absolute;
    top: -8px;
    right: -8px;
    width: 16px;
    height: 16px;
    border-radius: 1000px;
    background-color: black;
    border: solid black 1px;
    pointer-events: initial;
    cursor: pointer;
}

.heatmap-pinned-tooltip-close svg {
    position: absolute;
    width: 100%;
    height: 100%;
    fill: white;
    stroke: white;
}

.heatmap-pinned-tooltip-pin {
    position: absolute;
    left: 0px;
    bottom: 0px;
    width: calc(var(--tooltip-offset-x) / 0.6);
    height: calc(var(--tooltip-offset-y) / 0.6);
    z-index: -1;
}

.heatmap-pinned-tooltip-pin svg {
    fill: black;
}

.heatmap-overlay {
    position: absolute;
    width: 100%;
    height: 100%;
    pointer-events: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    z-index: 0;
}

.heatmap-overlay-shade {
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: black;
    opacity: 0.4;
    z-index: -1;
}

.heatmap-overlay-message {
    margin: 5px;
    padding-block: 1em;
    padding-inline: 1.6em;
    text-align: center;
    font-size: 150%;
    font-weight: bold;
    background-color: white;
    border: solid black 1px;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.75);
}

.heatmap-marker {
    stroke: black;
    stroke-width: 4;
    fill: none;
}

.heatmap-marker-x,
.heatmap-marker-y {
    stroke: black;
    stroke-width: 2;
    fill: none;
}
    
      
    <!---->
  </style>
  

      
  <div
    style="width:500px;padding-left:10px;padding-right:10px;padding-top:0px;padding-bottom:0px;""
  >
    
        
    <div
      id="testId"
      style="width:480px;height:500px;z-index:1;display:none;""
    >
      <div
        class="heatmap-main-div"
        style="position: relative; width: 100%; height: 100%;"
      >
        <div
          class="heatmap-canvas-div"
          style="position: absolute; width: 100%; height: 100%;"
        >
          <canvas
            height="0"
            style="position: absolute; width: 100%; height: 100%;"
            width="0"
          />
          <svg
            style="position: absolute; width: 100%; height: 100%;"
          >
            <rect
              class="heatmap-marker-y"
              height="0"
              rx="1"
              ry="1"
              width="NaN"
              x="NaN"
              y="0"
            />
          </svg>
        </div>
      </div>
    </div>
    
      
  </div>
  
      
  <div
    id="testId_loading"
    style="display: none;"
  >
    
        
    <svg
      height="200px"
      preserveAspectRatio="xMidYMid"
      style="background: none;"
      viewBox="0 0 100 100"
      width="200px"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      
          
      <circle
        cx="75"
        cy="50"
        fill="#363a3c"
        r="6.39718"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.875s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="67.678"
        cy="67.678"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.75s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="50"
        cy="75"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.625s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="32.322"
        cy="67.678"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.5s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="25"
        cy="50"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.375s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="32.322"
        cy="32.322"
        fill="#363a3c"
        r="4.80282"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.25s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="50"
        cy="25"
        fill="#363a3c"
        r="6.40282"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.125s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="67.678"
        cy="32.322"
        fill="#363a3c"
        r="7.99718"
      >
        
              
        <animate
          attributeName="r"
          begin="0s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
        
    </svg>
    
      
  </div>
  
    
</nightingale-sequence-heatmap>
`;

exports[`nightingale-sequence-heatmap tests it should display the sequence heatmap correctly 1`] = `
<nightingale-sequence-heatmap
  display-end="-1"
  display-start="1"
  heatmapid="testId"
  height="500"
  highlight-color="#FFEB3B66"
  min-height="10"
  min-width="10"
  style="display: inline-block; line-height: 0;"
  width="500"
>
  <!---->
  
      
  <style>
    
        #
    <!---->
    testId
     {
          /** Position of bottom-left corner of tooltip box relative to the mouse position */
          --tooltip-offset-x: 5px;
          /** Position of bottom-left corner of tooltip box relative to the mouse position */
          --tooltip-offset-y: 8px;
        }
        .heatmap-marker-x {
          /** background: 
    <!---->
    #FFEB3B66
     !important; */
          fill: 
    <!---->
    rgb(255, 235, 59)
     !important;
          fill-opacity: 
    <!---->
    0.359
     !important;
          stroke-width: 0 !important;
        }
        .heatmap-marker-y {
          /** background: 
    <!---->
    #FFEB3B66
     !important; */
          fill: 
    <!---->
    rgb(255, 235, 59)
     !important;
          fill-opacity: 
    <!---->
    0.359
     !important;
          stroke-width: 0 !important;
        }
        
    <!---->
    
.heatmap-main-div {
    font-family: sans-serif;
}

.heatmap-canvas-div {
    /* Set background-color here to change the background of the heatmap */
    background-color: none;
}

.heatmap-tooltip-box,
.heatmap-pinned-tooltip-box {
    z-index: 0;
    /* Avoid tooltip flickering */
    pointer-events: none;
}

.heatmap-tooltip-content,
.heatmap-pinned-tooltip-content {
    margin-left: var(--tooltip-offset-x);
    margin-bottom: var(--tooltip-offset-y);
    border: solid black 1px;
    padding-block: 0.35em;
    padding-inline: 0.7em;
    background-color: white;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.75);
    pointer-events: initial;
    width: max-content;
    line-height: 1;
}

.heatmap-pinned-tooltip-close {
    position: absolute;
    top: -8px;
    right: -8px;
    width: 16px;
    height: 16px;
    border-radius: 1000px;
    background-color: black;
    border: solid black 1px;
    pointer-events: initial;
    cursor: pointer;
}

.heatmap-pinned-tooltip-close svg {
    position: absolute;
    width: 100%;
    height: 100%;
    fill: white;
    stroke: white;
}

.heatmap-pinned-tooltip-pin {
    position: absolute;
    left: 0px;
    bottom: 0px;
    width: calc(var(--tooltip-offset-x) / 0.6);
    height: calc(var(--tooltip-offset-y) / 0.6);
    z-index: -1;
}

.heatmap-pinned-tooltip-pin svg {
    fill: black;
}

.heatmap-overlay {
    position: absolute;
    width: 100%;
    height: 100%;
    pointer-events: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    z-index: 0;
}

.heatmap-overlay-shade {
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: black;
    opacity: 0.4;
    z-index: -1;
}

.heatmap-overlay-message {
    margin: 5px;
    padding-block: 1em;
    padding-inline: 1.6em;
    text-align: center;
    font-size: 150%;
    font-weight: bold;
    background-color: white;
    border: solid black 1px;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.75);
}

.heatmap-marker {
    stroke: black;
    stroke-width: 4;
    fill: none;
}

.heatmap-marker-x,
.heatmap-marker-y {
    stroke: black;
    stroke-width: 2;
    fill: none;
}
    
      
    <!---->
  </style>
  

      
  <div
    style="width:500px;padding-left:10px;padding-right:10px;padding-top:0px;padding-bottom:0px;""
  >
    
        
    <div
      id="testId"
      style="width:480px;height:500px;z-index:1;display:none;""
    >
      <div
        class="heatmap-main-div"
        style="position: relative; width: 100%; height: 100%;"
      >
        <div
          class="heatmap-canvas-div"
          style="position: absolute; width: 100%; height: 100%;"
        >
          <canvas
            height="100"
            style="position: absolute; width: 100%; height: 100%;"
            width="100"
          />
          <svg
            style="position: absolute; width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
    
      
  </div>
  
      
  <div
    id="testId_loading"
    style="display: none;"
  >
    
        
    <svg
      height="200px"
      preserveAspectRatio="xMidYMid"
      style="background: none;"
      viewBox="0 0 100 100"
      width="200px"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      
          
      <circle
        cx="75"
        cy="50"
        fill="#363a3c"
        r="6.39718"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.875s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="67.678"
        cy="67.678"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.75s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="50"
        cy="75"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.625s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="32.322"
        cy="67.678"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.5s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="25"
        cy="50"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.375s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="32.322"
        cy="32.322"
        fill="#363a3c"
        r="4.80282"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.25s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="50"
        cy="25"
        fill="#363a3c"
        r="6.40282"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.125s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="67.678"
        cy="32.322"
        fill="#363a3c"
        r="7.99718"
      >
        
              
        <animate
          attributeName="r"
          begin="0s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
        
    </svg>
    
      
  </div>
  
    
</nightingale-sequence-heatmap>
`;

exports[`nightingale-sequence-heatmap tests it should zoom in 1`] = `
<nightingale-sequence-heatmap
  display-end="4"
  display-start="2"
  heatmapid="testId"
  height="500"
  highlight-color="#FFEB3B66"
  min-height="10"
  min-width="10"
  style="display: inline-block; line-height: 0;"
  width="500"
>
  <!---->
  
      
  <style>
    
        #
    <!---->
    testId
     {
          /** Position of bottom-left corner of tooltip box relative to the mouse position */
          --tooltip-offset-x: 5px;
          /** Position of bottom-left corner of tooltip box relative to the mouse position */
          --tooltip-offset-y: 8px;
        }
        .heatmap-marker-x {
          /** background: 
    <!---->
    #FFEB3B66
     !important; */
          fill: 
    <!---->
    rgb(255, 235, 59)
     !important;
          fill-opacity: 
    <!---->
    0.359
     !important;
          stroke-width: 0 !important;
        }
        .heatmap-marker-y {
          /** background: 
    <!---->
    #FFEB3B66
     !important; */
          fill: 
    <!---->
    rgb(255, 235, 59)
     !important;
          fill-opacity: 
    <!---->
    0.359
     !important;
          stroke-width: 0 !important;
        }
        
    <!---->
    
.heatmap-main-div {
    font-family: sans-serif;
}

.heatmap-canvas-div {
    /* Set background-color here to change the background of the heatmap */
    background-color: none;
}

.heatmap-tooltip-box,
.heatmap-pinned-tooltip-box {
    z-index: 0;
    /* Avoid tooltip flickering */
    pointer-events: none;
}

.heatmap-tooltip-content,
.heatmap-pinned-tooltip-content {
    margin-left: var(--tooltip-offset-x);
    margin-bottom: var(--tooltip-offset-y);
    border: solid black 1px;
    padding-block: 0.35em;
    padding-inline: 0.7em;
    background-color: white;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.75);
    pointer-events: initial;
    width: max-content;
    line-height: 1;
}

.heatmap-pinned-tooltip-close {
    position: absolute;
    top: -8px;
    right: -8px;
    width: 16px;
    height: 16px;
    border-radius: 1000px;
    background-color: black;
    border: solid black 1px;
    pointer-events: initial;
    cursor: pointer;
}

.heatmap-pinned-tooltip-close svg {
    position: absolute;
    width: 100%;
    height: 100%;
    fill: white;
    stroke: white;
}

.heatmap-pinned-tooltip-pin {
    position: absolute;
    left: 0px;
    bottom: 0px;
    width: calc(var(--tooltip-offset-x) / 0.6);
    height: calc(var(--tooltip-offset-y) / 0.6);
    z-index: -1;
}

.heatmap-pinned-tooltip-pin svg {
    fill: black;
}

.heatmap-overlay {
    position: absolute;
    width: 100%;
    height: 100%;
    pointer-events: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    z-index: 0;
}

.heatmap-overlay-shade {
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: black;
    opacity: 0.4;
    z-index: -1;
}

.heatmap-overlay-message {
    margin: 5px;
    padding-block: 1em;
    padding-inline: 1.6em;
    text-align: center;
    font-size: 150%;
    font-weight: bold;
    background-color: white;
    border: solid black 1px;
    box-shadow: 0px 5px 15px rgba(0, 0, 0, 0.75);
}

.heatmap-marker {
    stroke: black;
    stroke-width: 4;
    fill: none;
}

.heatmap-marker-x,
.heatmap-marker-y {
    stroke: black;
    stroke-width: 2;
    fill: none;
}
    
      
    <!---->
  </style>
  

      
  <div
    style="width:500px;padding-left:10px;padding-right:10px;padding-top:0px;padding-bottom:0px;""
  >
    
        
    <div
      id="testId"
      style="width:480px;height:500px;z-index:1;display:none;""
    >
      <div
        class="heatmap-main-div"
        style="position: relative; width: 100%; height: 100%;"
      >
        <div
          class="heatmap-canvas-div"
          style="position: absolute; width: 100%; height: 100%;"
        >
          <canvas
            height="0"
            style="position: absolute; width: 100%; height: 100%;"
            width="0"
          />
          <svg
            style="position: absolute; width: 100%; height: 100%;"
          />
        </div>
      </div>
    </div>
    
      
  </div>
  
      
  <div
    id="testId_loading"
    style="display: none;"
  >
    
        
    <svg
      height="200px"
      preserveAspectRatio="xMidYMid"
      style="background: none;"
      viewBox="0 0 100 100"
      width="200px"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      
          
      <circle
        cx="75"
        cy="50"
        fill="#363a3c"
        r="6.39718"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.875s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="67.678"
        cy="67.678"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.75s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="50"
        cy="75"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.625s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="32.322"
        cy="67.678"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.5s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="25"
        cy="50"
        fill="#363a3c"
        r="4.8"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.375s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="32.322"
        cy="32.322"
        fill="#363a3c"
        r="4.80282"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.25s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="50"
        cy="25"
        fill="#363a3c"
        r="6.40282"
      >
        
              
        <animate
          attributeName="r"
          begin="-0.125s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
          
      <circle
        cx="67.678"
        cy="32.322"
        fill="#363a3c"
        r="7.99718"
      >
        
              
        <animate
          attributeName="r"
          begin="0s"
          dur="1s"
          repeatCount="indefinite"
          times="0;0.1;0.2;0.3;1"
          values="4.8;4.8;8;4.8;4.8"
        />
        
          
      </circle>
      
        
    </svg>
    
      
  </div>
  
    
</nightingale-sequence-heatmap>
`;
